kratos/processes/tetrahedral_mesh_orientation_check.cpp

```diff
@@ -136,15 +136,15 @@ void TetrahedralMeshOrientationCheck::Execute()
                 if(it_face != faces_map.end()) { // It was actually found!!
                     // Mark the condition as visited. This will be useful for a check at the endif
                     std::vector<Condition::Pointer>& list_conditions = it_face->second;
-                    for (Condition::Pointer p_cond : list_conditions) {
+                    for (const Condition::Pointer& p_cond : list_conditions) {
                         p_cond->Set(VISITED,true);
                     }
 
                     if(mrOptions.Is(ASSIGN_NEIGHBOUR_ELEMENTS_TO_CONDITIONS)) {
                         GlobalPointersVector< Element > vector_of_neighbours;
                         vector_of_neighbours.resize(1);
                         vector_of_neighbours(0) = Element::WeakPointer( *it_elem.base() );
-                        for (Condition::Pointer p_cond : list_conditions) {
+                        for (const Condition::Pointer& p_cond : list_conditions) {
```

Comment:
so in here we were doing sa copy of the pointers?

Reply:
Of smart pointers in this case. But now I am thinking that the Condition::Pointer is an intrusive_ptr... Maybe we can keep it as it is

Reply:
I will wait to @RiccardoRossi to confirm if this improves (I would say yes)

Reply:
copying inteueive_ptrs is as expensive as for nor al smart_ptrs, so i wpuld expect this imprpves...nevertheless please give me the time to have a look

Reply:
@RiccardoRossi can I proceed with the merge?

Reply:
@RiccardoRossi can I proceed?

Reply:
@RiccardoRossi ping